p2p/host/autonat/autonat.go

```diff
@@ -286,10 +286,18 @@ func (as *AmbientAutoNAT) scheduleProbe() time.Duration {
 // Update the current status based on an observed result.
 func (as *AmbientAutoNAT) recordObservation(observation autoNATResult) {
 	currentStatus := as.status.Load()
+
+	// Ignore public observations with no address
+	if observation.Reachability == network.ReachabilityPublic && observation.address == nil {
```

Comment:
If I understand correctly, this can't really happen. A successful observation will always contain a multiaddress, and if the peer sends a multiaddress that we can't decode, we'd record this as a `ReachabilityUnknown` observation. We shouldn't do that either, we should discard that measurement.

Reply:
I checked. It cannot happen. cli.DialBack(ctx, pi.ID) ensures that address is not nil when err is nil. 
I added this because TestAutoNATObservationRecording specifically has a check for this case. 

I'll remove both.

Reply:
fixed.